src/models/users/matrix.ts

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export class MatrixUser {
+    private _localpart: string;
+    public readonly host: string;
+
+    /**
+     * Construct a Matrix user.
+     * @param userId The userId of the user.
+     * @param data Serialized data values
+     * @param escape Escape the user's localpart. Modify {@link MatrixUser~ESCAPE_DEFAULT}
+     *               to change the default value.
+     */
+    constructor(public userId: string, public readonly _data: Record<string, unknown> = {}, escape=MatrixUser.ESCAPE_DEFAULT) {
```

Comment:
The leading `_` suggests that it was always meant to be private.
```suggestion
    constructor(public userId: string, private readonly _data: Record<string, unknown> = {}, escape=MatrixUser.ESCAPE_DEFAULT) {
```